postgres/postgres.go

```diff
@@ -465,26 +493,71 @@ func (ds Datastore) UpdateFilter(isAuthenticated bool, filter *models.Filter) er
 		return err
 	}
 
-	updateStatement, err := updateStatement(filter)
-	if err != nil {
-		return err
+	rowsAffected := int64(0)
+
+	if filter.State != "" {
+		results, err := tx.Stmt(ds.updateFilterState).Exec(filter.FilterID, filter.State)
+		if err != nil {
+			if err = tx.Rollback(); err != nil {
+				return err
+			}
+			return err
+		}
+
+		rowsAffected, err = results.RowsAffected()
+		if err != nil {
+			if err = tx.Rollback(); err != nil {
+				return err
+			}
+			return err
+		}
 	}
 
-	stmt, err := tx.Prepare(updateStatement)
-	if err != nil {
-		return err
+	if !isAuthenticated && rowsAffected == 0 {
+		return errors.New("Bad request")
```

Comment:
Possibly 403, but external user might send in a bad json body full of variables we don't know about which would require a 400.

Solution might be to add validation in model to check the body sent into this endpoint and then this can be changed to 403.